torch/hub.py

```diff
@@ -628,9 +629,18 @@ def download_url_to_file(url: str, dst: str, hash_prefix: Optional[str] = None,
     # We deliberately save it in a temp file and move it after
     # download is complete. This prevents a local working checkpoint
     # being overridden by a broken download.
+    # We deliberately do not use NamedTemporaryFile to avoid restrictive
+    # file permissions being applied to the downloaded file.
     dst = os.path.expanduser(dst)
-    dst_dir = os.path.dirname(dst)
-    f = tempfile.NamedTemporaryFile(delete=False, dir=dst_dir)
+    for seq in range(tempfile.TMP_MAX):
+        tmp_dst = dst + '.' + uuid.uuid4().hex + '.partial'
```

Comment:
IIUC the risk of collision is close enough to zero that it can be ignored in practice... can we get rid of the `for-loop` logic and just raise if the file already exists (i.e. just keep line 643)?

Reply:
Hi @NicolasHug, this code mimics how `mkstemp` from cpython works. See here: https://github.com/python/cpython/blob/4bbf071635504ee1c6d44f99c98e3fad191a3b13/Lib/tempfile.py#L251-L267. Is there a particular reason why doing the same in PyTorch would be a problem?